src/AppBundle/Controller/UserAdminController.php

```diff
@@ -96,27 +95,13 @@ public function showUsersByDepartmentAction(Department $department)
 
     public function deleteUserByIdAction(User $user)
     {
-        // If Non-ROLE_HIGHEST_ADMIN try to delete user in other department
-        if (!$this->isGranted(Roles::ADMIN) && $user->getDepartment() !== $this->getUser()->getDepartment) {
-            throw new BadRequestHttpException();
-        }
-        try {
-            // This deletes the given user
+        if ($this->isGranted(ROLES::ADMIN) || $user->getDepartment() == $this->getUser()->getDepartment()) {
```

Comment:
Her burde det kastes en AccessDeniedException:
```php
throw $this->createAccessDeniedException();
```

Det er flere fordeler ved å kaste en exception i stedet for å bare videresende brukeren:
- Vi får en notification på Slack om at en bruker prøver å utføre en handling han ikke har tilgang til
- Brukervennlighet: brukeren får beskjed om han ikke har tilgang og trenger derfor ikke å bli forvirret over hvorfor brukeren ikke har blitt slettet. Han kan da spørre om å få oppgradert rettighetene sine hvis han egentlig skal ha høyere rettigheter.
- Det samme gjelder for debugging: En exception gir mer informasjon om hva som egentlig skjer, og man blir ikke forvirret av at det virker som alt fungerer riktig (blir videresent til useradmin page), men at brukeren ikke har blitt slettet